engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/IndexSchemaFilter.java

```diff
@@ -118,29 +118,41 @@ public boolean isPathIncluded(String relativePath) {
 	 * @return {@code true} if the path is included, {@code false} otherwise.
 	 */
 	private boolean isPathIncludedInternal(int relativeDepth, String relativePath,
-			boolean markAsEncountered, boolean includedByChild) {
-		boolean includedByThis = depthFilter.isEveryPathIncludedAtDepth( relativeDepth )
-				|| pathFilter.isExplicitlyIncluded( relativePath );
+			boolean markIncludedAsEncountered, boolean includedByChild) {
+		boolean includedByThis = ( depthFilter.isEveryPathIncludedAtDepth( relativeDepth )
+				|| pathFilter.isExplicitlyIncluded( relativePath ) )
+				&& !pathFilter.isExplicitlyExcluded( relativePath );
 
 		boolean includedByParent = true;
 		/*
 		 * The parent can filter out paths that are considered as included by a child,
 		 * by reducing the includeDepth in particular,
 		 * but it cannot include paths that are filtered out by a child.
+		 *
+		 * Note: If we only go into recursion when `includedByThis == true` path tracking of parent nodes will not record
+		 * the route.
 		 */
-		if ( parent != null ) {
+		if ( includedByThis && parent != null ) {
```

Comment:
Shouldn't this be:

```suggestion
		if ( includedByThis && includedByChild && parent != null ) {
```

... ?

Though now that I'm thinking of this, I wonder if the `includedByChild` parameter is even necessary anymore. We only ever call this method with `includedByChild == true`, don't we?

Reply:
Hmm, maybe ... I'll double-check it (and if the other map is still necessary)  😃 making changes to these paths is like playing Minesweeper 💣 😆

Reply:
sooooooooo it looks like, map is not needed, `includedByChild` is not needed 😃 
Thanks for looking into this 🥇😃